src/filemap.rs

```diff
@@ -36,16 +36,95 @@ pub fn write_all_files(file_map: &FileMap,
                        config: &Config)
                        -> Result<(HashMap<String, String>), io::Error> {
     let mut result = HashMap::new();
+    output_heading(mode).ok();
+
     for filename in file_map.keys() {
         let one_result = try!(write_file(&file_map[filename], filename, mode, config));
         if let Some(r) = one_result {
             result.insert(filename.clone(), r);
         }
     }
+    output_trailing(mode).ok();
 
     Ok(result)
 }
 
+pub fn output_heading(mode: WriteMode) -> Result<(), io::Error> {
```

Comment:
How could they go inside `write_file`? The header/footer needs to be output once. From my understanding `write_file` is invoked once per file. I could place them within `write_all_files` though.